public_html/wp-content/mu-plugins/latest-site-hints.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace WordCamp\Latest_Site_Hints;
+use function WordCamp\Sunrise\get_top_level_domain;
+
+use const WordCamp\Sunrise\{ PATTERN_YEAR_DOT_CITY_DOMAIN_PATH, PATTERN_CITY_SLASH_YEAR_DOMAIN_PATH };
+
+defined( 'WPINC' ) || die();
+
+add_action( 'wp', __NAMESPACE__ . '\maybe_add_latest_site_hints' );
+
+/**
+ * If user or bot visits WordCamp site that has newer site for the same city,
+ * add some hints for guiding them visit the latest site.
+ */
+function maybe_add_latest_site_hints() {
+	// Check if there is newer site for the WordCamp.
+	$latest_site = get_latest_site();
+	if ( ! $latest_site ) {
+		return;
+	}
+
+	// Hook in before `WordPressdotorg\SEO\Canonical::rel_canonical_link()`, so that callback can be removed.
+	add_action( 'wp_head', __NAMESPACE__ . '\canonical_link_past_home_pages_to_current_year', 9 );
+
+	/**
+	 * Show notification about newer WordCamp. Logged in users most probably know that already,
+	 * so no need to bother them (and we also misuse the body.admin-bar class).
+	 */
+	if ( ! is_user_logged_in() ) {
+		add_filter( 'body_class',   __NAMESPACE__ . '\add_notification_classes_to_body' );
+		add_action( 'wp_head',      __NAMESPACE__ . '\add_notification_styles' );
+		add_action( 'wp_body_open', __NAMESPACE__ . '\show_notification_about_latest_site' );
+	}
+}
+
+/**
+ * Add a `<link rel="canonical" ...` tag to the front page of past WordCamps, which points to the current year.
+ *
+ * This helps search engines know to direct queries for "WordCamp Seattle" to `seattle.wordcamp.org/2020`
+ * instead of `seattle.wordcamp.org/2019`, even if `/2019` has a higher historic rank.
+ */
+function canonical_link_past_home_pages_to_current_year() {
+	// We don't want to penalize historical content, we just want to boost the new site.
+	if ( ! is_front_page() ) {
+		return;
+	}
+
+	$latest_site = get_latest_site();
+	if ( ! $latest_site ) {
+		return;
+	}
+
+	// Remove default canonical link, to avoid duplicates.
+	// @todo: This will need to be updated if rel_canonical_link() is ever merged to Core.
+	remove_action( 'wp_head', 'WordPressdotorg\SEO\Canonical\rel_canonical_link' );
+
+	printf(
+		'<link rel="canonical" href="%s" />' . "\n",
+		esc_url( set_url_scheme( trailingslashit( '//' . $latest_site->domain . $latest_site->path ) ) )
+	);
+}
+
+/**
+ * If notification is shown, misuse body.admin-bar class as most of the WordCamp themes already that that into
+ * account. This makes its a whole lot easier to place the notification on top of contens.
+ */
+function add_notification_classes_to_body( $classes ) {
+	$classes[] = 'admin-bar'; // Add this as most of the current WordCamp themes already take the adminbar into account.
+	return $classes;
+}
+
+/**
+ * Simple styles for the notification.
+ */
+function add_notification_styles() { ?>
+	<style type="text/css">
+		html {
+			margin-top: 35px !important;
```

Comment:
```suggestion
			margin-top: var(--wp-admin--admin-bar--height) !important;
```

That was recently added to Core, and will automatically update from `32px` to `46px` at the various breakpoints.

Reply:
The idea is good, although the `--wp-admin--admin-bar--height` is defined in `admin-bar.css`, which isn't enqueued in the public site when user is logged out. If we end up loading that stylesheet, then the use of that variable is definitely wise.